core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -348,6 +348,9 @@ public int register(String subject,
         if (schemaId >= 0) {
           schema.setId(schemaId);
         } else {
+          while (guidToSchemaKey.containsKey(nextAvailableSchemaId)) {
```

Comment:
we would hit his scenario only when SR is in a bad state. Would it be better to just reject the request?